examples/cloudml-churn-prediction/trainer/trainer/task.py

```diff
@@ -0,0 +1,217 @@
+    parser.add_argument(
+        '--layer-sizes-scale-factor',
+        help="""Determine how the size of the layers in the DNN decays.
+        If value = 0 then the provided --hidden-units will be taken as is.""",
+        default=0.78535296735911486,
+        type=float,
+    )
+    parser.add_argument(
+        '--learning-rate',
+        help='Learning rate',
+        default=0.021388123321319803,
+        type=float,
+    )
+    parser.add_argument(
+        '--threshold',
+        help='Confidence Score threshold to resolve class predictions.',
+        default=0.34157646920229945,
+        type=float,
+    )
+    parser.add_argument(
+        '--checkpoint_steps',
+        help='Number of steps between each checkpoint.',
+        default=1000,
+        type=int,
+    )
+    return parser.parse_args(argv)
+
+
+def train_and_evaluate(flags):
+    """Runs model training and evaluation using TF Estimator API"""
+
+    #Get TF transform metadata generated during preprocessing
+    tf_transform_output = tft.TFTransformOutput(flags.input_dir)
+
+    #Define training spec
+    feature_spec = tf_transform_output.transformed_feature_spec()
+    train_input_fn = functools.partial(
+        input_util.input_fn,
+        input_dir=flags.input_dir,
+        mode=tf.estimator.ModeKeys.TRAIN,
+        batch_size=flags.train_batch_size,
+        num_epochs=flags.num_epochs,
+        label_name=metadata.LABEL_COLUMN,
+        feature_spec=feature_spec
+    )
+    train_spec = tf.estimator.TrainSpec(
+        train_input_fn, max_steps=flags.train_steps)
+
+    #Define eval spec
+    eval_input_fn = functools.partial(
+        input_util.input_fn,
+        input_dir=flags.input_dir,
+        mode=tf.estimator.ModeKeys.EVAL,
+        batch_size=flags.eval_batch_size,
+        num_epochs=1,
+        label_name=metadata.LABEL_COLUMN,
+        feature_spec=feature_spec
+    )
+
+    exporter = tf.estimator.FinalExporter(
+        "export", functools.partial(
+            input_util.tfrecord_serving_input_fn,
+            feature_spec=feature_spec,
+            label_name=metadata.LABEL_COLUMN))
+
+    eval_spec = tf.estimator.EvalSpec(
+        eval_input_fn,
+        steps=flags.eval_steps,
+        start_delay_secs=flags.eval_start_secs,
+        exporters=[exporter],
+        name='churn-eval'
+    )
+
+    #Define training config
+    run_config = tf.estimator.RunConfig(
+        save_checkpoints_steps=flags.checkpoint_steps,
+        tf_random_seed=metadata.SEED,
+        model_dir=flags.job_dir
+    )
+
+    #Build the estimator
+    feature_columns = model.get_feature_columns(
+        tf_transform_output, exclude_columns=metadata.NON_FEATURE_COLUMNS)
+    num_intervals = metadata.NUM_INTERVALS
+    estimator = model.build_estimator(
+        run_config, flags, feature_columns, num_intervals)
+
+    #Run training and evaluation
+    tf.estimator.train_and_evaluate(estimator, train_spec, eval_spec)
+
+
+def main():
+    #Parse command-line arguments
```

Comment:
FYI try to be [more sparse](http://go/pystyle#385-block-and-inline-comments) with comments. Also comments should have 1 space after the `#` if they start on a new line.